File: app/src/main/java/com/vershininds/mixture/sample/presentation/modules/rx2interactor/router/SampleRx2Router.java
package com.vershininds.mixture.sample.presentation.modules.rx2interactor.router;

import android.app.Activity;
import android.content.Intent;
import android.support.annotation.NonNull;

import javax.inject.Inject;

import com.vershininds.mixture.sample.data.SampleObject;
import com.vershininds.mixture.sample.presentation.modules.details.DetailsModuleInput;
import com.vershininds.mixture.sample.presentation.modules.rx2interactor.contract.SampleRx2RouterContract;
import com.vershininds.mixture.router.AbstractRouter;
import com.vershininds.mixture.view.AndroidComponent;


public class SampleRx2Router extends AbstractRouter<SampleRx2RouterContract.Presenter>
        implements SampleRx2RouterContract.Router {

    private DetailsModuleInput detailsModuleInput;

    @Inject
    public SampleRx2Router(DetailsModuleInput detailsModuleInput) {
        this.detailsModuleInput = detailsModuleInput;
    }

    @Override
    public void showDetailsScreen(@NonNull AndroidComponent androidComponent, @NonNull SampleObject data) {
        Activity activity = androidComponent.getActivity();
        Intent intent = detailsModuleInput.createActivityIntent(activity, data);
        activity.startActivity(intent);
    }
}

File: app/src/main/java/com/vershininds/mixture/sample/presentation/modules/rx2interactor/router/SampleRx2Router.kt
package com.vershininds.mixture.sample.presentation.modules.rx2interactor.router

import com.vershininds.mixture.router.AbstractRouter
import com.vershininds.mixture.router.MvpRouter
import com.vershininds.mixture.sample.presentation.modules.details.DetailsModuleContract
import com.vershininds.mixture.sample.presentation.modules.rx2interactor.contract.SampleRx2RouterContract
import com.vershininds.mixture.sample.presentation.modules.rx2interactor.contract.SampleRx2RouterContract.TypeRouterAction
import com.vershininds.mixture.view.AndroidComponent
import javax.inject.Inject


class SampleRx2Router @Inject constructor(private val detailsModuleContract: DetailsModuleContract) :
        AbstractRouter<MvpRouter.Listener>(), SampleRx2RouterContract.Router {

    override fun actionHandler(androidComponent: AndroidComponent, action: TypeRouterAction) {
        when (action) {
            is TypeRouterAction.DetailsScreenAction -> showDetailsScreen(androidComponent, action.params)
        }
    }

    override fun showDetailsScreen(androidComponent: AndroidComponent, params: SampleRx2RouterContract.DetailsParams) {
        val activity = androidComponent.activity
        val intent = detailsModuleContract.createActivityIntent(activity, params.sampleObject)
        activity.startActivity(intent)
    }
}